resolver/metric_resolver.go

```diff
@@ -0,0 +1,67 @@
+package resolver
+
+import (
+	"context"
+	"net"
+
+	"github.com/Shopify/goose/statsd"
+)
+
+type metricResolver struct {
+	resolver Resolver
+	timer    *statsd.Timer
+	tags     statsd.Tags
+}
+
+func NewMetricResolver(resolver Resolver, timer *statsd.Timer, tags statsd.Tags) Resolver {
```

Comment:
`NewMetricResolver` feels kind of confusing since it sounds like you are resolving metrics. Since this lives in the resolver package, what about something like `NewWithMetrics(...)` so that calling it would look like : 

```go
resolver.NewWithMetrics(...)
```